src/gxhash/platform/arm.rs

```diff
@@ -27,13 +27,17 @@ pub unsafe fn load_unaligned(p: *const State) -> State {
 
 #[inline(always)]
 pub unsafe fn get_partial_safe(data: *const State, len: usize) -> State {
-    // Temporary buffer filled with zeros
-    let mut buffer = [0i8; VECTOR_SIZE];
-    // Copy data into the buffer
-    core::ptr::copy(data as *const i8, buffer.as_mut_ptr(), len);
-    // Load the buffer into a __m256i vector
-    let partial_vector = vld1q_s8(buffer.as_ptr());
-    vaddq_s8(partial_vector, vdupq_n_s8(len as i8))
+    // // Temporary buffer filled with zeros
+    // let mut buffer = [0i8; VECTOR_SIZE];
+    // // Copy data into the buffer
+    // core::ptr::copy(data as *const i8, buffer.as_mut_ptr(), len);
+    // // Load the buffer into a __m256i vector
+    // let partial_vector = vld1q_s8(buffer.as_ptr());
+    // vaddq_s8(partial_vector, vdupq_n_s8(len as i8))
+
+    let indices = vld1q_s8([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15].as_ptr());
+    let mask = vreinterpretq_s8_u8(vcgtq_s8(vdupq_n_s8(len as i8), indices));
+    std::intrinsics::simd::simd_masked_load(mask, data as *const i8, vdupq_n_s8(len as i8))
```

Comment:
So do I understand correctly that it would be *valid* for LLVM to basically emit the same assembly as before, but somehow it doesn't?

I'm not a codegen expert so I have no idea whether this is just hard for LLVM to do or something they could reasonably fix. Maybe it'd be worth making an LLVM bugreport about this? (We can try to find some people that could help nail down the core issue here, if you are interested.)